lib/THC/THCGeneral.c

```diff
@@ -23,6 +23,22 @@ void THCudaInit(THCState* state)
   THCudaCheck(cudaGetDevice(&device));
 
   state->rngState = (THCRNGState*)malloc(sizeof(THCRNGState));
+
+#ifdef USE_CNMEM
+  cnmemDevice_t devices[count];
+  double multiplier = (getenv("CUTORCH_CNMEM")) ? (atof(getenv("CUTORCH_CNMEM"))) : (0.95);
+  for (int i = 0; i < count; ++i) {
+    devices[i].device = i;
+    size_t free_mem, used_mem;
+    cudaMemGetInfo(&free_mem, &used_mem);
+
+    devices[i].size = free_mem * multiplier;
+    devices[i].numStreams = 0;
+    devices[i].streams = NULL;
+  }
+  THCnmemCheck(cnmemInit(count, devices, CNMEM_FLAGS_DEFAULT));
```

Comment:
could you briefly explain this? 
Does this mean that 95% of the free memory will now be preallocated to CNMem? 
If so, that seems like a very bad thing to do in a setting where users are sharing GPUs. Something that increases the Heap of memory tracked by CNMem based on average usage seems more reasonable. Like how we do heaptracking for garbage collection: https://github.com/torch/cutorch/blob/master/lib/THC/THCGeneral.c#L595-L658
cc: @colesbury and @wickedfoo for comments